### PR TITLE
HIVE-24455. Fix broken junit framework in storage-api.

### PR DESCRIPTION
**Pull request by omalley**
Update the storage-api surefire plugin version to match the rest of hive.

### CODE REVIEW

File: storage-api/pom.xml [hunk 1]
```diff
@@ -177,6 +177,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-plugin</artifactId>
+        <version>3.0.0-M4</version>
```
Comment (jcamachor) [inline storage-api/pom.xml:180]: Should we add this version in the properties block too?

Comment (omalley) [reply storage-api/pom.xml:180]: We could, but it is only used in the single place.

Comment (omalley) [reply storage-api/pom.xml:180]: I mostly use properties for versions that get referenced multiple times, but if you want me to change it I will.

Comment (jcamachor) [reply storage-api/pom.xml:180]: I don't think it's a major issue, I just find it easier to have all versions together in the properties block and I also thought this was the approach taken by Hive in general (I do not think this is formalized anywhere, just based on previous changes made to the top pom.xml file). I'll leave it up to you.

**Review by jcamachor — APPROVED** [review]

**Comment by omalley** [comment]
I fixed this with 539291f2574.

**Pull request closed without merging**